packages/roosterjs-editor-dom/test/selections/deleteSelectedContentTest.ts

```diff
@@ -0,0 +1,141 @@
+import createRange from '../../lib/selection/createRange';
+import deleteSelectedContent from '../../lib/selection/deleteSelectedContent';
+import getHtmlWithSelectionPath from '../../lib/selection/getHtmlWithSelectionPath';
+import setHtmlWithSelectionPath from '../../lib/selection/setHtmlWithSelectionPath';
+
+describe('deleteSelectedContent', () => {
+    let div: HTMLDivElement;
+    beforeEach(() => {
+        div = document.createElement('div');
+        div.contentEditable = 'true';
+        document.body.appendChild(div);
+    });
+
+    afterEach(() => {
+        document.body.removeChild(div);
+        div = null;
+    });
+
+    function runTest(html: string, expected: string) {
+        const range = setHtmlWithSelectionPath(div, html);
+        const position = deleteSelectedContent(div, range);
+        const result = getHtmlWithSelectionPath(div, createRange(position));
```

Comment:
@JiuqingSong , could you confirm that the loop in getHtmlWithSelectionPath is working appropriately?

I could not get the test file attached to the debugger, but I added some logs:
![image](https://user-images.githubusercontent.com/15259586/93943376-3f6d6400-fce7-11ea-9225-6663ee573fd0.png)

and when I insert a table like you have in the Whole Table 2 example:
aa<table><tbody><tr><td>line1</td><td>line2</td></tr><tr><td>line3</td><td>line4</td></tr></tbody></table>bb<!--{"start":[1],"end":[2]}-->

I am getting that after the TBODY child the next sibling is NULL. Not sure why that is happening but it might be related to your test below where you don't get the expected result in the selection path...

![image](https://user-images.githubusercontent.com/15259586/93943350-2fee1b00-fce7-11ea-8291-ca75c42ffc92.png)

Reply:
Oh I get it now :) The body does not have siblings. I am sleepy... But then the loop is not working as intended because it never gets to check the row, right?

Reply:
@aniagonzalez 
The code is to make sure there is TBODY under TABLE

In most cases, when we insert a table, there is no TBODY between TABLE and TR because it is optional. But this causes a problem when we get/set html with selection path. Because when we set HTML directly into container and the HTML doesn't have TBODY, broder will add TBODY for us, this causes the DOM structure changed, and selection path becomes invalid. To mitigate, we need to always add TBODY when we get HTML with selection. So that when set HTML, no extra tags will be added. In your case, there is already TBODY, so that it won't get TR and nothing need to be added, which is an expected behavior.

Reply:
Thank you, that clears things up a lot! I am learning so much just by going through your CR :)